### PR TITLE
#184651331 Add Login API Enpoint

### PR DESCRIPTION
**Pull request by bstwagir**
#### What does this PR do?
Add log in API endpoint

#### Description of Task to be completed?

Have the following endpoint working
POST /api/login

#### How should this be manually tested?

============================ PRE-REQUIRED: =================================
====> .env

USER_SECRET_KEY= YOUR JWT SECRET KEY 
NODE_ENV= ENV YOU WANT YOU USE FROM YOUR CONFIG
DB_USERNAME= YOUR POSTGRES USERNAME
DB_PASSWORD= YOUR POSTGRES PASSWORD
DB_DEVELOPMENTDB=YOUR DATABASE NAME
DB_HOST=YOUR HOSTNAME

====> Needed
User in the database

====>
- Clone the repository 
- Navigate to the project 
- In the terminal type: npm install to install the required dependencies.
* RUN npm start
* Using Postman test the endpoint above

#### What are the relevant pivotal tracker/Trello stories?
#184651331

### CODE REVIEW

**Review by ruberwa — DISMISSED** [review]

**Review by ruberwa — DISMISSED** [review]
Good

**Review by bahati10 — DISMISSED** [review]

**Review by kagame8732 — DISMISSED** [review]
well designed !

**Review by JulesNIYOMWUNGERI — DISMISSED** [review]

**Review by Raphaela664 — DISMISSED** [review]

**Review by ruberwa — COMMENTED** [review]
- You have unresolved Conflict
- Test Coverage are bellow minimum

File: .gitignore
Comment (nshutip) [inline .gitignore:1]: you deleted the right line of code, and the new line added in wrong.

Comment (nshutip) [inline .gitignore:1]: The deleted line was written in a correct format, while the new added line is incorect.
It should be written as "**/coverage"

File: .sequelizerc
Comment (nshutip) [inline .sequelizerc:1]: line 1 is ES5, I believe it should be changed to ES6

**Review by nshutip — COMMENTED** [review]
Well implemented!

**Review by Raphaela664 — DISMISSED** [review]

**Review by heshima18 — DISMISSED** [review]

**Review by nshutip — DISMISSED** [review]
Well implemented!

Points to note:
- The login tests could have also been put in their specific test file (login.test.js or auth.test.js) instead of being in app.test.js

**Review by ruberwa — DISMISSED** [review]
Functionalities are good!

But this :

- One test is failing in circleci and It may be that it has expired because it's being copied and pasted into the test.
- In the PR message, It's not clear how to test it locally, Specifically the protected part!
- There is some Unnecessary Eslint disable

**Review by ruberwa — APPROVED** [review]
Functionalities are good!

But these :

- In the PR message, It's not clear how to test it locally, Specifically the protected part!

- Your test is not following the same structure we are using as you did on your Controller and Router

**Review by Raphaela664 — APPROVED** [review]

**Pull request merged**